modules/common/src/test/java/com/ibm/cloud/cloudant/internal/SdkTimeoutTestNew.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * © Copyright IBM Corporation 2020. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.ibm.cloud.cloudant.internal;
+
+import com.ibm.cloud.sdk.core.http.ServiceCall;
+import com.ibm.cloud.sdk.core.security.BasicAuthenticator;
+import com.ibm.cloud.sdk.core.security.NoAuthAuthenticator;
+import okhttp3.*;
+import okhttp3.internal.connection.RealCall;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Field;
+import java.util.concurrent.TimeUnit;
+
+import static org.testng.Assert.assertEquals;
+
+
+@PrepareForTest({RealCall.class})
+public class SdkTimeoutTestNew {
+    final long defaultTimeoutValue = 150L;
+    final long customTimeoutValue = 10L;
+
+    // Every test case tests an authenticator.
+    final long testCases[] = {
+            // 1. case: its default timeout value checked,
+            defaultTimeoutValue,
+            // 2. case: a custom overwrite possibility is checked.
+            customTimeoutValue
+    };
+
+    @Test
+    void testNoAuthTimeout() throws NoSuchFieldException, IllegalAccessException {
+        CloudantBaseService myService = new CloudantBaseService(null, new NoAuthAuthenticator()) {};
+        myService.setServiceUrl("https://cloudant.example");
+
+        // Create a server request for testing
+        HttpUrl requestUrl = HttpUrl.parse(myService.getServiceUrl()).newBuilder().addPathSegment("/").build();
+        Request req = new Request.Builder().url(requestUrl).get().build();
+
+        for (int i = 0; i < 2; i++) {
```

Comment:
It would be not so confusing if we could remove the for loop and the array, and use helper methods between the assertions. This way the helper methods could be reused in the `testBasicAuthTimeout` test case too. 
I sketched what I was thinking about (the code might be not perfect though).

```java
// create service call for testing method call
RealCall requestCall = getRealCall(req, myService);
// assert
assertEquals(TimeUnit.MILLISECONDS.toSeconds(requestCall.getClient().readTimeoutMillis()), defaultTimeoutValue);

// change timeout
setClientReadTimeout(customTimeoutValue, myService);

// create service a new call for testing
RealCall requestCall = getRealCall(req, myService)
// assert
assertEquals(TimeUnit.MILLISECONDS.toSeconds(requestCall.getClient().readTimeoutMillis()), customTimeoutValue);

...
RealCall getRealCall(Request req, CloudantBaseService service){
  ServiceCall<RealCall> serviceCall = service.createServiceCall(req, null);
  Field callField = serviceCall.getClass().getDeclaredField("call");
  callField.setAccessible(true);
  return (RealCall) callField.get(serviceCall);
}

void setClientReadTimeout(long timeout, CloudantBaseService service){
  OkHttpClient client = service.getClient();
  client = client.newBuilder().readTimeout(timeout, TimeUnit.SECONDS).build();
  service.setClient(client);
}
```

Reply:
I have introduced some helper function: 9caad49

I would like to keep the for cycle since it complies with `node` and `python` testcase structure.
Additionally, `IamAuth` and `SessionAutht` needs different handling. 
Could you please review it again? WDYT about the current structure.

Reply:
As I have written under the node PR: https://github.com/IBM/cloudant-node-sdk/pull/488#discussion_r720181085